packages/build-utils/src/convert-runtime-to-plugin.ts

```diff
@@ -229,8 +300,39 @@ export function convertRuntimeToPlugin(
 
       await fs.ensureDir(dirname(nft));
       await fs.writeFile(nft, json);
+
+      // Extend the list of directories and files that were created by the
+      // Legacy Runtime with the list of directories and files that were
+      // created for the entrypoint that was just processed above.
+      newPathsRuntime = new Set([
+        ...newPathsRuntime,
+        ...newFilesEntrypoint,
+        ...newDirectoriesEntrypoint,
+      ]);
     }
 
+    // Instead of of waiting for all of the linking to be done for every
+    // entrypoint before processing the next one, we immediately handle all
+    // of them one after the other, while then waiting for the linking
+    // to finish right here, before we clean up newly created files below.
+    await Promise.all(linkersRuntime);
```

Comment:
Do you need a semaphore to prevent unbounded File IO?

Seems like this will fail when lambdaFiles is in the thousands and the file descriptor limit is reached.

Reply:
I think you wrote the code for this, correct? I'm only adding a `Promise.all` to wait until all of them are finished.

Previously, the program would continue, even if all the linking wasn't done yet (which was fine before).

Reply:
Regardless, I do agree that it's potentially a good idea to add, but I think it's unrelated to the change in this PR and can be done in a separate one.